soc: stm32: move stack size definitions to the correct place

djiatsaf-st opened the pull request:
Following discussions in https://github.com/zephyrproject-rtos/zephyr/issues/85368 

This PR moves the definitions of `MAIN_STACK_SIZE`, `IDLE_STACK_SIZE`, and `ISR_STACK_SIZE` to a family Kconfig level.

These changes specifically concern the STM32 families  boards with low RAM memory (less than 8 KiB).

djiatsaf-st committed:
soc: st: stm32f0x: remove kernel stack size definitions

These kconfig symbols will be handle in family Kconfig
level.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
soc: st: stm32g0x: remove kernel stack size definitions

These kconfig symbols will be handle in family Kconfig
level

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
soc: st: stm32l0x: remove kernel stack size definitions

These kconfig symbols will be handle in family Kconfig
Level

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

mathieuchopstm reviewed (commented):
LGTM functionally.

erwango commented on soc/st/stm32/stm32f0x/Kconfig.defconfig:
```diff
@@ -17,4 +17,14 @@ config TASK_WDT_HW_FALLBACK_DELAY
 	depends on TASK_WDT_HW_FALLBACK
 	default 100
 
+# Reduce kernel stack sizes to fit in 8 KiB RAM
+config MAIN_STACK_SIZE
+	default 640 if SOC_STM32F030X4 || SOC_STM32F030X8 || SOC_STM32F051X8
+
+config IDLE_STACK_SIZE
+	default 200 if SOC_STM32F030X4 || SOC_STM32F030X8 || SOC_STM32F051X8
+
+config ISR_STACK_SIZE
+	default 512 if SOC_STM32F030X4 || SOC_STM32F030X8 || SOC_STM32F051X8
```
Let's work on a common Kconfig piece that would define profiles based on DT_CHOSEN(zephyr_ram) and apply it on the whole STM32 family.

mathieuchopstm commented:
@djiatsaf-st @erwango is this intended to ship in 4.3? otherwise, update milestone

djiatsaf-st committed:
soc: st: stm32: add common kconfig symbols for kernel stack size

Provide default values for common kconfig symbols for main, idle and
isr stack sizes, which apply on all STM32 MCU families with low RAM
memory(less than 8 KiB).
These kconfig symbols help reduce kernel sizes to fit within
limited RAM.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

erwango reviewed (changes requested):
`soc: st: stm32: add common kconfig symbols for kernel stack size` should be the first commit.
Also:
> add common kconfig symbol for main, idle and isr stack sizes,

Please reword the message: You're not adding the Kconfig symbol, as it exists already, but provide default value depending on ...

mathieuchopstm commented on soc/st/stm32/Kconfig.defconfig:
```diff
@@ -20,6 +20,21 @@ config CLOCK_CONTROL
 config CORTEX_M_SYSTICK
 	default n if STM32_LPTIM_TIMER
 
+# Reduce kernel stack sizes to fit into RAM with less than 8 KiB
```
```suggestion
# Reduce kernel stack sizes to fit MCUs with 8 KiB RAM or less
```

etienne-lms commented on soc/st/stm32/Kconfig.defconfig:
```diff
@@ -20,6 +20,21 @@ config CLOCK_CONTROL
+
+config IDLE_STACK_SIZE
+	default 100 if SRAM_SIZE <= 2
+	default 150 if SRAM_SIZE <= 4
```
Despite this change is synced with the value previosly used in applicable SoCs, I find it strange the stack size is not a multiple of a 32bit word size.
This comment is outside the scope of this P-R, not blocking.
